Index: gcc/testsuite/gcc.target/aarch64/pr70044.c
===================================================================
--- a/src/gcc/testsuite/gcc.target/aarch64/pr70044.c	(revision 234974)
+++ b/src/gcc/testsuite/gcc.target/aarch64/pr70044.c	(nonexistent)
@@ -1,14 +0,0 @@
-/* { dg-do link } */
-/* { dg-require-effective-target lto } */
-/* { dg-options "-flto -O --save-temps -fno-omit-frame-pointer" } */
-
-extern int atoi (const char *);
-
-int
-main (int argc, char **argv)
-{
-  return atoi (argv[0]) + 1;
-}
-
-/* Check that the frame pointer really is created.  */
-/* { dg-final { scan-lto-assembler "add	x29, sp," } } */
Index: gcc/config/aarch64/aarch64.c
===================================================================
--- a/src/gcc/config/aarch64/aarch64.c	(revision 234974)
+++ b/src/gcc/config/aarch64/aarch64.c	(revision 234968)
@@ -6917,26 +6917,10 @@
 static void
 aarch64_override_options_after_change (void)
 {
-  /* The logic here is that if we are disabling all frame pointer generation
-     then we do not need to disable leaf frame pointer generation as a
-     separate operation.  But if we are *only* disabling leaf frame pointer
-     generation then we set flag_omit_frame_pointer to true, but in
-     aarch64_frame_pointer_required we return false only for leaf functions.
-
-     PR 70044: We have to be careful about being called multiple times for the
-     same function.  Once we have decided to set flag_omit_frame_pointer just
-     so that we can omit leaf frame pointers, we must then not interpret a
-     second call as meaning that all frame pointer generation should be
-     omitted.  We do this by setting flag_omit_frame_pointer to a special,
-     non-zero value.  */
-
-  if (flag_omit_frame_pointer == 2)
-    flag_omit_frame_pointer = 0;
-
   if (flag_omit_frame_pointer)
     flag_omit_leaf_frame_pointer = false;
   else if (flag_omit_leaf_frame_pointer)
-    flag_omit_frame_pointer = 2;
+    flag_omit_frame_pointer = true;
 
   /* If not optimizing for size, set the default
      alignment to what the target wants */
