--- a/src/gcc/testsuite/ChangeLog
+++ b/src/gcc/testsuite/ChangeLog
@@ -1,14 +1,6 @@
 2016-03-18  Andre Vieira  <andre.simoesdiasvieira@arm.com>
 
 	Backport from mainline
-	2015-06-09  James Greenhalgh  <james.greenhalgh@arm.com>
-
-	* g++.dg/ext/pr57735.C: Do not override -mfloat-abi directives
-	passed by the testsuite driver.
-
-2016-03-18  Andre Vieira  <andre.simoesdiasvieira@arm.com>
-
-	Backport from mainline
 	2015-07-16  Kyrylo Tkachov  <kyrylo.tkachov@arm.com>
 
 	* gcc.target/arm/no-volatile-in-it.c: Skip if -mcpu is overriden.
--- a/src/gcc/testsuite/g++.dg/ext/pr57735.C
+++ b/src/gcc/testsuite/g++.dg/ext/pr57735.C
@@ -1,7 +1,4 @@
 /* { dg-do compile { target arm*-*-* } } */
-/* { dg-require-effective-target arm_arch_v5te_ok } */
-/* { dg-require-effective-target arm_arm_ok } */
-/* { dg-skip-if "do not override -mfloat-abi" { *-*-* } { "-mfloat-abi=*" } {"-mfloat-abi=soft" } } */
 /* { dg-options "-march=armv5te -marm  -mtune=xscale -mfloat-abi=soft -O1" } */
 
 typedef unsigned int size_t;
