--- a/src/gcc/testsuite/gcc.target/arm/no-volatile-in-it.c
+++ b/src/gcc/testsuite/gcc.target/arm/no-volatile-in-it.c
@@ -1,6 +1,5 @@
 /* { dg-do compile } */
 /* { dg-require-effective-target arm_thumb2_ok } */
-/* { dg-skip-if "do not override -mcpu" { *-*-* } { "-march=*" "-mcpu=*" } { "-mcpu=cortex-m7" } } */
 /* { dg-options "-Os -mthumb -mcpu=cortex-m7" } */
 
 int
--- a/src/gcc/testsuite/ChangeLog
+++ b/src/gcc/testsuite/ChangeLog
-2016-03-18  Andre Vieira  <andre.simoesdiasvieira@arm.com>
-
-	Backport from mainline
-	2015-07-16  Kyrylo Tkachov  <kyrylo.tkachov@arm.com>
-
-	* gcc.target/arm/no-volatile-in-it.c: Skip if -mcpu is overriden.
-
