--- a/src/gcc/config/aarch64/aarch64.c	2017-07-11 17:41:55.803147289 +0000
+++ b/src/gcc/config/aarch64/aarch64.c	2017-07-11 17:43:46.575579185 +0000
@@ -8956,7 +8956,8 @@ aarch64_classify_symbol (rtx x, rtx offs
 	  /* This is alright even in PIC code as the constant
 	     pool reference is always PC relative and within
 	     the same translation unit.  */
-	  if (CONSTANT_POOL_ADDRESS_P (x))
+	  if (!aarch64_pcrelative_literal_loads
+	      && CONSTANT_POOL_ADDRESS_P (x))
 	    return SYMBOL_SMALL_ABSOLUTE;
 	  else
 	    return SYMBOL_FORCE_TO_MEM;
