# DP: Fix go/67589, return typ of main

Index: gcc-5-5.2.1/src/libgo/runtime/go-main.c
===================================================================
--- gcc-5-5.2.1.orig/src/libgo/runtime/go-main.c
+++ gcc-5-5.2.1/src/libgo/runtime/go-main.c
@@ -38,7 +38,7 @@ main (int argc, char **argv)
   runtime_isarchive = false;
 
   if (runtime_isstarted)
-    return NULL;
+    return 0;
   runtime_isstarted = true;
 
   runtime_check ();
